File: Niveau 2/2 – Découverte des tableaux/07 - Journée des cadeaux.py
nbHabitant = int(input())
fortunes = [0] * nbHabitant
for i in range(nbHabitant):
    fortunes[i] = int(input())
fortunes.sort()
if(nbHabitant % 2 != 0):
    print(fortunes[nbHabitant // 2])
else:
    i = nbHabitant//2
    print((fortunes[i-1] + fortunes[i])/2) 

File: Niveau 2/2 – Découverte des tableaux/07 - Journée des cadeaux.cpp
#include <iostream>
#include <algorithm>
#include <cmath>
using namespace std;
int main()
{
  int nbHabitants;
  cin >> nbHabitants;
  int fortunes[nbHabitants] = {0};
  for (int i = 0; i < nbHabitants; i++)
  {
    int f;
    cin >> f;
    fortunes[i] = f;
  }
  sort(&fortunes[0], &fortunes[nbHabitants]);
  if (nbHabitants % 2 == 0)
  {
    double a = (double)fortunes[nbHabitants / 2];
    double b = (double)fortunes[(nbHabitants / 2) - 1];
    cout << ((a + b) / 2);
  }
  else
  {
    cout << fortunes[(nbHabitants / 2)];
  }
}